qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDatabaseQueries.groovy

```diff
@@ -342,31 +342,83 @@ class CustomerDatabaseQueries {
         connection.setAutoCommit(false)
 
         connection.withCloseable {it ->
             try {
-                int customerId = createNewCustomer(it, customer)
-                storeAffiliation(it, customerId, customer.affiliations)
+                createNewAffiliation(it, affiliation)
                 connection.commit()
             } catch (Exception e) {
                 log.error(e.message)
                 log.error(e.stackTrace.join("\n"))
                 connection.rollback()
                 connection.close()
-                throw new DatabaseQueryException("Could not create customer.")
+                throw new DatabaseQueryException("Could not create affiliation.")
             }
 
         }
-        */
+    }
+
+    private boolean affiliationExists(Affiliation affiliation) {
+        String query = "SELECT * FROM affililation WHERE organization = ? " +
+                "AND address_addition=? " +
+                "AND street=? " +
+                "AND country=? " +
+                "AND postal_code=? " +
+                "AND city=? " +
+                "AND category=?"
+
+        Connection connection = databaseSession.getConnection()
+
+        def affiliationAlreadyInDb = false
+
+        connection.withCloseable {
+            def statement = connection.prepareStatement(query)
+            statement.setString(1, affiliation.organisation)
+            statement.setString(2, affiliation.addressAddition)
+            statement.setString(3, affiliation.street)
+            statement.setString(4, affiliation.country)
+            statement.setString(5, affiliation.postalCode)
+            statement.setString(6, affiliation.city)
+            statement.setString(7, affiliation.category.toString())
+            statement.execute()
+            def result = statement.getResultSet()
+            affiliationAlreadyInDb = result.next()
+        }
+        return affiliationAlreadyInDb
+    }
+
+    private static int createNewAffiliation(Connection connection, Affiliation affiliation) {
+        String query = "INSERT INTO affiliation (organization, address_addition, street, country, postal_code, city, category) " +
+                "VALUES(?, ?, ?, ?, ?, ?, ?)"
+
+        List<Integer> generatedKeys = []
+
+        def statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)
+        statement.setString(1, affiliation.organisation)
+        statement.setString(2, affiliation.addressAddition)
+        statement.setString(3, affiliation.street)
+        statement.setString(4, affiliation.country)
+        statement.setString(5, affiliation.postalCode)
+        statement.setString(6, affiliation.city)
+        statement.setString(7, affiliation.category.toString())
+        statement.execute()
+        def keys = statement.getGeneratedKeys()
+        while (keys.next()){
+            generatedKeys.add(keys.getInt(1))
+        }
 
+        return generatedKeys[0]
```

Comment:
why do you iterate over all generated keys only to return the first one found?

Reply:
This was also present in the implementation of the Customer creation method for the SQL database. I kept it for consistency sake. I think the idea was to see if null or a  specific value is returned to see if the creation of an affiliation was successful.